File: 66-plus-one/plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        List<Integer> result = Arrays.stream(digits).boxed().collect(Collectors.toList());
        Collections.reverse(result);
        int add = 1, i = 0;
        while (add > 0){
            if (i < result.size()){
                if (result.get(i) == 9){
                    result.set(i, 0);
                }
                else{
                    result.set(i, result.get(i) + 1);
                    add = 0;
                }
            }
            else{
                result.add(1);
                add = 0;
            }
            i += 1;
        }
        Collections.reverse(result);
        return result.stream().mapToInt(j->j).toArray();
    }
}

File: 66-plus-one/plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        digits.reverse()
        add, i = 1,  0
        while add > 0:
            if i < len(digits):
                if digits[i] == 9:
                    digits[i] = 0
                else:
                    add = 0
                    digits[i] += 1
            else:
                digits.append(1)
                add = 0
            i += 1
        digits.reverse()
        return digits